File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02732/p02732_num4_parsed.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

int main() {
    int N;
    std::cin >> N;

    std::vector<int> A(N);
    for (int i = 0; i < N; i++) {
        std::cin >> A[i];
    }

    std::unordered_map<int, int> count;
    for (int i = 0; i < N; i++) {
        count[A[i]]++;
    }

    for (int k = 0; k < N; k++) {
        int ans = N - 1 - count[A[k]] + 1;
        std::cout << ans << std::endl;
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02732/p02732_num4_parsed.py
def find_number_of_ways(N, balls):
    frequency_count = {}
    for ball in balls:
        if ball in frequency_count:
            frequency_count[ball] += 1
        else:
            frequency_count[ball] = 1

    answer = []
    for k in range(1, N + 1):
        count = 0
        if balls[k - 1] in frequency_count:
            count = frequency_count[balls[k - 1]] - 1
        answer.append(count)

    return answer


# Read input
N = int(input())
balls = list(map(int, input().split()))

# Solve problem
answer = find_number_of_ways(N, balls)

# Print output
for ans in answer:
    print(ans)